pkg/discovery/worker/compliance/affinity_processor.go

```diff
@@ -0,0 +1,170 @@
+	commManager *AffinityCommodityManager
+
+	nodes []*api.Node
+	pods  []*api.Pod
+}
+
+func NewAffinityProcessor(config *affinityProcessorConfig) (*AffinityProcessor, error) {
+	allNodes, err := config.k8sClusterScraper.GetAllNodes()
+	if err != nil {
+		return nil, err
+	}
+	allPods, err := config.k8sClusterScraper.GetAllPods()
+	if err != nil {
+		return nil, err
+	}
+	return &AffinityProcessor{
+		ComplianceProcessor: NewComplianceProcessor(),
+		commManager:         NewAffinityCommodityManager(),
+
+		nodes: allNodes,
+		pods:  allPods,
+	}, nil
+}
+
+// TODO if there is an error, fail the whole discovery? currently, error is handle in place and won't affect other discovery results.
+func (am *AffinityProcessor) ProcessAffinityRules(entityDTOs []*proto.EntityDTO) []*proto.EntityDTO {
+	am.GroupEntityDTOs(entityDTOs)
+	for _, pod := range am.pods {
+		am.processAffinityPerPod(pod)
+	}
+	return am.GetAllEntityDTOs()
+}
+
+func (am *AffinityProcessor) processAffinityPerPod(pod *api.Pod) {
+	affinity := pod.Spec.Affinity
+	if affinity == nil {
+		return
+	}
+
+	nodeMap := make(map[string]*api.Node)
+	for _, currNode := range am.nodes {
+		nodeMap[currNode.Name] = currNode
+	}
+
+	podsNodesMap := make(map[*api.Pod]*api.Node)
+	for _, currPod := range am.pods {
+		hostingNode, exist := nodeMap[currPod.Spec.NodeName]
+		if !exist || hostingNode == nil {
+			continue
+		}
+		podsNodesMap[currPod] = hostingNode
+	}
+
+	nodeSelectorTerms := getAllNodeSelectors(affinity)
+	nodeAffinityAccessCommoditiesSold, nodeAffinityAccessCommoditiesBought, err := am.commManager.GetAccessCommoditiesForNodeAffinity(nodeSelectorTerms)
+	if err != nil {
+		glog.Errorf("Failed to build commodity: %s", err)
+		return
+	}
+
+	podAffinityTerms := getAllPodAffinityTerms(affinity)
+	podAffinityCommodityDTOsSold, podAffinityCommodityDTOsBought, err := am.commManager.GetAccessCommoditiesForPodAffinityAntiAffinity(podAffinityTerms)
+	if err != nil {
+		glog.Errorf("Failed to build commodity for pod affinity: %s", err)
+		return
+	}
+
+	for _, node := range am.nodes {
+		if matchesNodeSelector(pod, node) && matchesNodeAffinity(pod, node) {
+			am.addAffinityAccessCommodities(pod, node, nodeAffinityAccessCommoditiesSold, nodeAffinityAccessCommoditiesBought)
+		}
+		if interPodAffinityMatches(pod, node, podsNodesMap) {
+			am.addAffinityAccessCommodities(pod, node, podAffinityCommodityDTOsSold, podAffinityCommodityDTOsBought)
+		}
+	}
+}
+
+func getAllNodeSelectors(affinity *api.Affinity) []api.NodeSelectorTerm {
+	nodeSelectorTerms := []api.NodeSelectorTerm{}
+	// TODO we only parse RequiredDuringSchedulingIgnoredDuringExecution for now.
+	if affinity.NodeAffinity != nil && affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		nodeSelectorTerms = append(nodeSelectorTerms, affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms...)
+	}
+	//glog.Infof("node selectors are %++v", nodeSelectorTerms)
+	return nodeSelectorTerms
+}
```

Comment:
A minor suggestion - since this is just a single if, not a loop, there isn't a need to append.  Maybe it's clearer to return directly as in the following:

```
if affinity.NodeAffinity == nil || affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
    return []api.NodeSelectorTerm{}
}

return affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms...
```